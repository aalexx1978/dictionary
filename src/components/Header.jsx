import React, { Component } from "react";
import { Navbar } from "react-bootstrap";
import logo from "../logo.png";
class Header extends Component {
    render() {
        return (
            <Navbar collapseOnSelect>
                <Navbar.Header>
                    <Navbar.Brand>
                        <img src={logo} alt="logo" className="App-logo" />
                        Dictionary
                    </Navbar.Brand>
                    <Navbar.Toggle />
                </Navbar.Header>
            </Navbar>
        );
    }
}

export default Header;
