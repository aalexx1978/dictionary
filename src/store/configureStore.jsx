import { createStore, applyMiddleware, compose } from "redux";
import rootReducer from "../reducers/allReducers";
import thunk from "redux-thunk";
import { logger } from "redux-logger";
const store = createStore(
    rootReducer,
    window.__REDUX_DEVTOOLS_EXTENSION__ &&
        window.__REDUX_DEVTOOLS_EXTENSION__(),
    compose(applyMiddleware(thunk, logger))
);

export default function configureStore() {
    return store;
}
