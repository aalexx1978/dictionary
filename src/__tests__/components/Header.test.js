import React from "react";
import Enzyme from "enzyme";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import toJson from "enzyme-to-json";
Enzyme.configure({ adapter: new Adapter() });
import configureStore from "redux-mock-store";

import Header from "../../components/Header";
describe("<Header />", () => {
    describe("render()", () => {
        test("renders the component", () => {
            const wrapper = shallow(<Header />);
            const component = wrapper.dive();

            expect(toJson(component)).toMatchSnapshot();
        });
    });
});
