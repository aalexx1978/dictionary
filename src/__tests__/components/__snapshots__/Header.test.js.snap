// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render() renders the component 1`] = `
<Navbar
  bsClass="navbar"
  bsStyle="default"
  collapseOnSelect={true}
  componentClass="nav"
  fixedBottom={false}
  fixedTop={false}
  fluid={false}
  inverse={false}
  onToggle={[Function]}
  staticTop={false}
>
  <NavbarHeader>
    <NavbarBrand>
      <img
        alt="logo"
        className="App-logo"
        src="logo.png"
      />
      Dictionary
    </NavbarBrand>
    <NavbarToggle />
  </NavbarHeader>
</Navbar>
`;
